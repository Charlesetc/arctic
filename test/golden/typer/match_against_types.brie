define main [
  let x = One 1
  x = Two 1 2
  x =  Three 1 2 3
  match x [
    One x -> x
    Two a b -> b
    Three a b c -> c
  ]
]
