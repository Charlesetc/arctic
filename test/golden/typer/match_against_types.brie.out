root[(ident(define):: ident(main):: block[][(ident(define):: ident(x):: (ident(One):: ident(1)::Integer)::[^ One Integer, Two Integer Integer, Three Integer Integer Integer ])::, (ident(_update):: ident(x)::[^ One Integer ] (ident(Two):: ident(1)::Integer ident(2)::Integer)::[^ Two Integer Integer ])::[^ One Integer, Two Integer Integer ], (ident(_update):: ident(x)::[^ One Integer, Two Integer Integer ] (ident(Three):: ident(1)::Integer ident(2)::Integer ident(3)::Integer)::[^ Three Integer Integer Integer ])::[^ One Integer, Two Integer Integer, Three Integer Integer Integer ], (ident(match):: ident(x)::[^ One Integer, Two Integer Integer, Three Integer Integer Integer ] block[][(ident(->):: (ident(One):: ident(x)::Integer):: (ident(x)::Integer)::Integer)::, (ident(->):: (ident(Two):: ident(a)::Integer ident(b)::Integer):: (ident(b)::Integer)::Integer)::, (ident(->):: (ident(Three):: ident(a)::Integer ident(b)::Integer ident(c)::Integer):: (ident(c)::Integer)::Integer)::]::)::Integer]::Function)::]::